crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -1421,3 +1402,32 @@ pub(crate) fn report_runtime_check_against_non_runtime_checkable_protocol(
     ));
     diagnostic.info("See https://docs.python.org/3/library/typing.html#typing.runtime_checkable");
 }
+
+pub(crate) fn report_attempted_protocol_instantiation(
+    context: &InferContext,
+    call: &ast::ExprCall,
+    protocol: ProtocolClassLiteral,
+) {
+    let Some(builder) = context.report_lint(&CALL_NON_CALLABLE, call) else {
+        return;
+    };
+    let db = context.db();
+    let class_name = protocol.name(db);
+    let mut diagnostic = builder.into_diagnostic(format_args!(
+        "Cannot instantiate abstract class `{class_name}`",
```

Comment:
Are you sure that it's clearest to use the term "abstract" for this? I can see why it's not unreasonable, but both mypy and pyright just say "Cannot instantiate protocol class 'MyProtocol'". It seems like introducing the term "abstract" might be unnecessary complexity, and evoke only-sort-of-related things like ABCMeta? It just forces us to explain below that inheriting Protocol makes a class "implicitly abstract", which seems unnecessary when we could just as well say "it inherits Protocol, making it a protocol class, and protocol classes can't be instantiated"

Reply:
Hmm, fair point. In my mental model, protocol classes are banned from being instantiated for exactly the same reason that ABCs with abstract methods are banned from being instantiated: both classes represent abstract interfaces to some extent (with the former really just being an extension of the latter in terms of how far you take that idea). But using the term "abstract" probably doesn't help our users, who won't necessary have the same mental model!